source/logrepl/cdc.go

```diff
@@ -113,35 +113,63 @@ func (i *CDCIterator) StartSubscriber(ctx context.Context) error {
 	return nil
 }
 
-// Next returns the next record retrieved from the subscription. This call will
-// block until either a record is returned from the subscription, the
-// subscription stops because of an error or the context gets canceled.
-// Returns error when the subscription has been started.
-func (i *CDCIterator) Next(ctx context.Context) (opencdc.Record, error) {
+// NextN takes and returns up to n records from the queue. NextN is allowed to
+// block until either at least one record is available or the context gets canceled.
+func (i *CDCIterator) NextN(ctx context.Context, n int) ([]opencdc.Record, error) {
 	if !i.subscriberReady() {
-		return opencdc.Record{}, errors.New("logical replication has not been started")
+		return []opencdc.Record{}, errors.New("logical replication has not been started")
```

Comment:
Now that we're returning a slice, we can return `nil` if we're returning an error.

```suggestion
		return nil, errors.New("logical replication has not been started")
```

Same for the other lines that return an error, also in the snapshot and combined iterators.